L58-python-async-api.md

```diff
@@ -0,0 +1,1752 @@
+            executing the next interceptor in chain or invoking the
+            actual RPC on the underlying Channel. It is the interceptor's
+            responsibility to call it if it decides to move the RPC forward.
+            The interceptor can use
+            `response_future = continuation(client_call_details, request)`
+            to continue with the RPC. `continuation` returns an object that is
+            both a Call for the RPC and a Future. In the event of RPC
+            completion, the return Call-Future's result value will be
+            the response message of the RPC. Should the event terminate
+            with non-OK status, the returned Call-Future's exception value
+            will be an RpcError.
+          client_call_details: A ClientCallDetails object describing the
+            outgoing RPC.
+          request: The request value for the RPC.
+
+        Returns:
+            An object that is both a Call for the RPC and a Future.
+            In the event of RPC completion, the return Call-Future's
+            result value will be the response message of the RPC.
+            Should the event terminate with non-OK status, the returned
+            Call-Future's exception value will be an RpcError.
+        """
+
+
+# grpc.aio.UnaryStreamClientInterceptor
+class UnaryStreamClientInterceptor:
+    """Affords intercepting unary-stream invocations."""
+
+    async def intercept_unary_stream(self,
+                               continuation: Callable[[grpc.ClientCallDetails, Request], grpc.aio.Call[AsyncIterable[Response]]],
+                               client_call_details: grpc.ClientCallDetails,
+                               request: Request
+        ) -> grpc.aio.Call[AsyncIterable[Response]]:
+        """Intercepts a unary-stream invocation.
+
+        Args:
+          continuation: A function that proceeds with the invocation by
+            executing the next interceptor in chain or invoking the
+            actual RPC on the underlying Channel. It is the interceptor's
+            responsibility to call it if it decides to move the RPC forward.
+            The interceptor can use
+            `response_iterator = continuation(client_call_details, request)`
+            to continue with the RPC. `continuation` returns an object that is
+            both a Call for the RPC and an iterator for response values.
+            Drawing response values from the returned Call-iterator may
+            raise RpcError indicating termination of the RPC with non-OK
+            status.
+          client_call_details: A ClientCallDetails object describing the
+            outgoing RPC.
+          request: The request value for the RPC.
+
+        Returns:
+            An object that is both a Call for the RPC and an iterator of
+            response values. Drawing response values from the returned
+            Call-iterator may raise RpcError indicating termination of
+ 
```

Comment:
Should we also mention the presence of new exception classes, like ``AioRpcError`` ?

Reply:
Good idea. I adding a section description for exceptions that on our road map: `grpc.BaseError`, `grpc.AbortError`, and `grpc.UsageError`.

As for `grpc.RpcError`, existing design of the `grpc.RpcError` class is empty (no methods / attributes). It generates confusion for many new gRPC users that it is unclear why the caught exception type doesn't contain any information. I hope at some point, we should promote the design of `AioRpcError` to `grpc.RpcError`.